org.metaborg.spoofax.shell.core/src/test/java/org/metaborg/spoofax/shell/core/ClassPathInterpreterLoaderTest.java

```
@@ -0,0 +1,341 @@
+                                                null },
+                                              { missingDynSemProperties(VFS.getManager()
+                                                  .resolveFile("ram:///rootDirTwo/")),
+                                                "Missing \"dynsem.properties\" file",
+                                                null },
+                                              { multipleLocations(VFS.getManager()
+                                                  .resolveFile("ram:///rootDirThree/")),
+                                                NO_EXCEPTION,
+                                                null },
+                                              { badWeatherLocations(VFS.getManager()
+                                                  .resolveFile("ram:///rootDirFour/")),
+                                                null, ClassNotFoundException.class } });
+
+    }
+
+    private static Supplier<List<FileObject>> goodWeatherLocations(FileObject rootDir)
+        throws FileSystemException {
+        return () -> {
+            try {
+                rootDir.createFolder();
+                FileObject dynSemProperties = rootDir.resolveFile("dynsem.properties");
+                dynSemProperties.createFile();
+                PrintWriter printWriter =
+                    new PrintWriter(dynSemProperties.getContent().getOutputStream());
+                printWriter.println("project.javapackage = org.metaborg.spoofax.shell.core");
+                printWriter
+                    .println("source.langname = ClassPathInterpreterLoaderTest$TestCPLoader");
+                printWriter.flush();
+                printWriter.close();
+                dynSemProperties.close();
+                List<FileObject> res = new ArrayList<FileObject>(1);
+                res.add(rootDir);
+                return res;
+            } catch (FileSystemException e) {
```

You can just throw these exceptions; when caught by JUnit they will automatically make the test fail.